utils/utilFunctions.ts

```diff
@@ -16,3 +13,22 @@ export async function sleep(timeoutTimeInMs: number) {
 export function randomBoolean() {
   return !Math.round(Math.random());
 }
+
+// CHECK THIS, onko mitään järkeä
+export async function invalidateSingleQueryKey(
+  queryClient: QueryClient,
+  queryKey: singleQueryKey,
+) {
+  await queryClient.invalidateQueries({
+    queryKey: [`${queryKey}`],
+  });
+}
+
+export async function invalidateMultipleQueryKeys(
+  queryClient: QueryClient,
+  queryKeyArray: multipleQueryKeys,
+) {
+  await queryClient.invalidateQueries({
+    queryKey: queryKeyArray,
+  });
+}
```

Comment:
Tämä on taas näitä omia ideoita :smile: Tuskin tässäkään mitään järkeä on, mutta halusin kokeilla

Eli tein oman funktion sekä yhdelle että monelle `queryKey`n poistamiselle. Näissä "hienona" juttuna olisi se, että TypeScript osaisi ehdottaa itse mitä mahdollisia `queryKey`-vaihtoehtoja on olemassa. Olettaen, että ne ovat merkitty `shared/types.ts`-tiedostoon

Mietin, että onko tässä jokin riski käyttää `queryClient`iä omassa tiedostossaan. Mutta kun otan `queryClient`in 
 parametrina Reactin componentista ei sitä riskiä pitäisi olla. Kysyin vielä varalta Gemini AI:lta huomaako hän jotain "riskiä" tässä, ei löytänyt :smile:

Reply:
Toimii kyllä noin kuten teit, mutta itse tekisin `QueryKeys` objektin `types.ts` tiedoston ohjeiden mukaan, jolloin ei tarvitse tämmöistä ylimääräistä funktiota mutta saa kuitenkin auto-completen käyttöön. Eli kun haluaa invalidoida queryn se onnistuu seuraavasti:

```ts
queryClient.invalidateQueries({ queryKey: QueryKeys.GIFTS });
```

Reply:
Hyvä idea! Toimii hyvin :+1: